core/src/kura.rs

```diff
@@ -193,9 +192,12 @@ impl Kura {
         kura: &Kura,
         mut shutdown_receiver: tokio::sync::oneshot::Receiver<()>,
     ) {
-        let (mut written_block_count, mut latest_block_hash) = {
-            let block_data_guard = kura.block_data.lock();
-            (block_data_guard.len(), block_data_guard.last().map(|d| d.0))
+        let mut written_block_count = kura.init_block_count;
+        let mut latest_written_block_hash = {
+            let block_data = kura.block_data.lock();
+            written_block_count
```

Comment:
Do I understand it right that this is basically the same behavior as before? Seeing how `init_block_count` gets set to `block_data.len()` in the init.. Or can something actually modify the `block_data` vec in the meantime?

UPD: I see now that you mention that `store_block` being called before or quickly after spawning the `receive_blocks_loop` thread, makes sense.